scripts/add_gen3_repo.py

```diff
@@ -0,0 +1,150 @@
+                        "See ap_verify_dataset_template/config for examples.",
+            **kwargs)
+        self.add_argument("--drop-gen2", action="store_true",
+                          help="Create a standalone Gen 3 repo instead of sharing files with Gen 2. "
+                               "Intended for use only once ap_verify no longer supports Gen 2.")
+
+
+def main():
+    args = _Parser().parse_args()
+    log = lsst.log.Log.getLogger("add_gen3_repo")
+
+    # To convert consistently, don't use any previous output
+    dataset = ap_verify.dataset.Dataset("test")
+    gen3_repo = os.path.join(dataset.datasetRoot, "preloaded")
+    if os.path.exists(gen3_repo):
+        log.warn("Clearing out %s and making it from scratch...", gen3_repo)
+        shutil.rmtree(gen3_repo)
+    os.makedirs(gen3_repo)
+
+    mode = "copy" if args.drop_gen2 else "relsymlink"
+
+    log.info("Converting calibs...")
+    with tempfile.TemporaryDirectory() as tmp:
+        workspace = ap_verify.workspace.Workspace(tmp)
+        ap_verify.ingestion.ingestDataset(dataset, workspace)
+
+        gen2_repo = workspace.dataRepo
+        gen2_calibs = workspace.calibRepo
+        # Files stored in the Gen 2 part of the dataset, can be safely linked
+        _migrate_gen2_to_gen3(dataset, gen2_repo, gen2_calibs, gen3_repo, mode,
+                              config_file="convertRepo_calibs.py")
+        # Our refcats and defects are temporary files, and must not be linked
+        _migrate_gen2_to_gen3(dataset, gen2_repo, gen2_calibs, gen3_repo, mode="copy",
+                              config_file="convertRepo_copied.py")
+
+    log.info("Exporting Gen 3 registry to configure new repos...")
+    _export_for_copy(dataset, gen3_repo)
+
+
+def _migrate_gen2_to_gen3(dataset, gen2_repo, gen2_calib_repo, gen3_repo, mode, config_file):
+    """Convert a Gen 2 repository into a Gen 3 repository.
+
+    Parameters
+    ----------
+    dataset : `lsst.ap.verify.dataset.Dataset`
+        The dataset being migrated.
+    gen2_repo, gen2_calib_repo : `str`
+       The locations of the original repositories.
+    gen3_repo : `str`
+       The location of the Gen 3 repository. Must exist, but need not be
+       initialized as a repository.
+    mode : {'relsymlink', 'copy'}
+       Whether the Gen 3 repo should contain symbolic links to the Gen 2
+       datasets, or an independent copy.
+    config_file : `str`
+       The config file (in the dataset config directory) with a configuration
+       for `~lsst.obs.base.gen2to3.ConvertRepoTask`
+    """
+    instrument = _get_instrument_class(dataset.camera)
+    config = os.path.join(dataset.configLocation, config_file)
+
+    # Call the script instead of calling ConvertRepoTask directly, to
+    # avoid manually having to do a lot of setup that may change in the future
+    # calib/<instrument>, refcats, and skymaps collections created by default
+    convert(gen3_repo, gen2_repo, instrument,
```

Comment:
I would rather this be `lsst.obs.base.script.convert`, since convert is a pretty generic word.